processor/sourcemap/processor.go

```diff
@@ -33,36 +35,31 @@ func NewProcessor() pr.Processor {
 	return &processor{schema}
 }
 
-func (p *processor) Validate(buf []byte) error {
+func (p *processor) Validate(raw interface{}) error {
 	validationCount.Inc()
 
-	var pa payload
-	err := json.Unmarshal(buf, &pa)
+	bytes, err := json.Marshal(raw.(map[string]interface{})["sourcemap"])
```

Comment:
this could panic with something like `sourcemap.NewProcessor().Validate(map[string]string{})`

Reply:
Right. Is this 16d630f695a6b1aeb35f8acfbef3f622b444cfcc better?

Reply:
looks great

Reply:
Cool, thanks!

Fixed the other issues you pointed out as well. Did you find anything else?